### DISCUSSION TITLE
Testing custom PAL

### DISCUSSION THREAD

**Discussion by Jatsekku** [openthread/openthread#9715] [Q&A]
I've created own version of PAL that should simulate hardware under posix (I'm aware of simulations PAL, but I wanted to learn how it's implemented). When I built it, it is handled for ot-cli-ftd example, but when I run ot-test-flash I see that it's not testing my flash implementation, how can I set up tests to use my custom PAL code?

**Comment by jwhui** [comment]
See the simulation platform implementation in [examples/platforms/simulation/flash.c](https://github.com/openthread/openthread/blob/main/examples/platforms/simulation/flash.c).
